components/link_checker/src/lib.rs

```diff
@@ -91,4 +126,22 @@ mod tests {
         assert!(res.code.is_none());
         assert!(res.error.is_some());
     }
+
+    #[test]
+    fn can_validate_anchors() {
+        let url = "https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect";
```

Comment:
can we skip the remote access and just test `check_page_for_anchor` for both tests?